### DISCUSSION TITLE
Using `useLingui` hook together with `useLingui` macro?

### DISCUSSION THREAD

**Discussion by jgarplind** [lingui/js-lingui#2118] [Q&A] [answered]
I have a small component like so:

```jsx
import { msg } from "@lingui/core/macro";
import { useLingui } from "@lingui/react";
import { otherMessageDescriptor } from "somewhere";

const Component= () => {
  const { _ } = useLingui();

  return (
<>
    <div>{_(msg`Locally defined string`)}</div>
    <div>{_(otherMessageDescriptor)}</div>
</>);
};
```

Now, if I wanted to use the new `useLingui`-macro in v5, I would replace the imports, and the first divs contents with:

```
t`Locally defined string`
```

But then how should I handle the legitimate use case for the regular `useLingui` hook in the div below?

Can the hook and macro even be combined?

**Comment by timofei-iatsenko** [comment]
use useLingui macro in both cases. Simply replace useLingui runtime with macro version

**Reply by jgarplind** [reply]
But the macro version doesn't expose `_` any more. Should I use `t` everywhere instead?

**Reply by timofei-iatsenko** [reply, marked as answer]
Yes, it's the same.

**Reply by jgarplind** [reply]
Thanks!

**Reply by timofei-iatsenko** [reply]
BTW, there is an automatic [codemod](https://gist.github.com/AndrewIngram/299a7370ac636478cc9600872d306031) migrating 

```jsx
import { msg } from '@lingui/macro';
import { useLingui } from '@lingui/react';

export default function MyComponent() {
  const { _ } = useLingui();

  return (
    <>
      <p>{_(msg`Hello again`)}</p>
    </>
  );
}
```

To `useLingui` macro 

```jsx
import { useLingui } from '@lingui/react/macro';

export default function MyComponent() {
  const { t } = useLingui();

  return (
    <>
      <p>{t`Hello again`}</p>
    </>
  );
}
```
